src/jdk.compiler/share/classes/com/sun/tools/javac/tree/TreeInfo.java

```diff
@@ -964,6 +966,41 @@ public static Name fullName(JCTree tree) {
         }
     }
 
+    public record SymAndTree(Symbol symbol, JCTree tree) {}
+
+    public static java.util.List<SymAndTree> symbolsFor(List<JCTree> nodes) {
+        java.util.List<SymAndTree> result = new ArrayList<>();
+        for (JCTree node : nodes) {
+            java.util.List<SymAndTree> partialResult = symbolsFor(node);
+            if (!partialResult.isEmpty()) {
+                result.addAll(partialResult);
+            }
+        }
+        return result;
+    }
+
+    public static java.util.List<SymAndTree> symbolsFor(JCTree node) {
```

Comment:
Can you provide some examples on what this method is supposed to do? E.g. example input and output?